lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.cpp

```diff
@@ -169,6 +173,72 @@ createTransform3d(const Ufe::SceneItem::Ptr& item, NextTransform3dFn nextTransfo
     return stackOps.empty() ? nextTransform3dFn() : UsdTransform3dMayaXformStack::create(usdItem);
 }
 
+// Class for setMatrixCmd() implementation.  Should be rolled into a future
+// command class with full undo / redo support
+class UsdSetMatrix4dUndoableCmd : public Ufe::SetMatrix4dUndoableCommand
+{
+public:
+    UsdSetMatrix4dUndoableCmd(
+        const Ufe::Path&     path,
+        const Ufe::Vector3d& oldT,
+        const Ufe::Vector3d& oldR,
+        const Ufe::Vector3d& oldS,
+        const Ufe::Matrix4d& newM)
+        : Ufe::SetMatrix4dUndoableCommand(path)
+        , _oldT(oldT)
+        , _oldR(oldR)
+        , _oldS(oldS)
+    {
+        // Decompose new matrix to extract TRS.  Neither GfMatrix4d::Factor
```

Comment:
Since now the parent -absolute code no longer decomposes the local matrix into TRS, we do it here in the plugin, using essentially the same code as what Maya used to use, namely MTransformationMatrix.